sklearn/preprocessing/data.py

```diff
@@ -1901,3 +1950,522 @@ def transform(self, X):
+
+        Parameters
+        ----------
+        X : sparse matrix CSC, shape (n_samples, n_features)
+            The data used to scale along the features axis. The sparse matrix
+            needs to be nonnegative.
+
+        inverse : bool, optional (default=False)
+            If False, apply forward transform. If True, apply
+            inverse transform.
+
+        Returns
+        -------
+        X : sparse matrix CSC, shape (n_samples, n_features)
+            Projected data.
+        """
+
+        for feature_idx in range(X.shape[1]):
+            column_slice = slice(X.indptr[feature_idx],
+                                 X.indptr[feature_idx + 1])
+            X.data[column_slice] = self._transform_col(
+                X.data[column_slice], self.quantiles_[:, feature_idx], inverse)
+
+        return X
+
+    def _check_inputs_transform(self, X):
+        """Private function to check the inputs before transforming"""
+        X = check_array(X, accept_sparse='csc', copy=self.copy,
+                        dtype=[np.float64, np.float32])
+        # we only accept positive sparse matrix when ignore_implicit_zeros is
+        # false
+        if (not self.ignore_implicit_zeros and
+                (sparse.issparse(X) and np.any(X.data < 0))):
+            raise ValueError('QuantileTransformer only accepts non-negative'
+                             ' sparse matrices')
+        check_is_fitted(self, 'quantiles_')
+        # check that the dimension of X are adequate with the fitted data
+        if X.shape[1] != self.quantiles_.shape[1]:
+            raise ValueError('X does not have the same number of feature than'
+                             ' the previously fitted data. Got {} instead of'
+                             ' {}'.format(X.shape[1],
+                                          self.quantiles_.shape[1]))
+        # check the output PDF
+        if self.output_distribution not in ('normal', 'uniform'):
+            raise ValueError("'output_distribution' has to be either 'norm' or"
+                             " 'uniform'. Got {} instead.".format(
+                                 self.output_distribution))
+
+        return X
+
+    def transform(self, X):
+        """Feature-wise transformation of the data.
+
+        Parameters
+        ----------
+        X : ndarray or sparse matrix, shape (n_samples, n_features)
+            The data used to scale along the features axis. If a sparse
+            matrix is provided, it will be converted into a sparse
+            ``csc_matrix``. Additionally, the sparse matrix needs to be
+            nonnegative if `ignore_implicit_zeros` is False.
+
+        Returns
+        -------
+        Xt : ndarray or sparse matrix, shape (n_samples, n_features)
+            The projected data.
+        """
+        X = self._check_inputs_transform(X)
+
+        if sparse.issparse(X):
+            return self._sparse_transform(X, inverse=False)
```

Comment:
Could we please inline these methods, and refactor with `_transform(X, inverse)`?